src/main/java/storages/CustomHashSet.java

```diff
@@ -0,0 +1,57 @@
+package storages;
+
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class CustomHashSet<E> {
+    private static final Object PRESENT = new Object();
+    private HashMap<E, Object> elements;
```

Comment:
do not use HashMap, instead use array, ArrayList ot LinkedList

Reply:
Fixed in https://github.com/kolosokvit/ticket-service/pull/3/commits/091895e1f96fc30220d1f6eb22b73efc0992bf45